libs/mrwebrtc-unityplugin/src/native_renderer.cpp

```diff
@@ -0,0 +1,309 @@
+    }
+    g_renderers.emplace(videoTrackHandle, std::make_shared<NativeRenderer>(videoTrackHandle));
+  }
+}
+
+void NativeRenderer::Destroy(mrsRemoteVideoTrackHandle videoTrackHandle) {
+  {
+    // Global lock
+    std::lock_guard guard(g_lock);
+    DestroyUnsafe(videoTrackHandle);
+  }
+}
+
+void NativeRenderer::DestroyUnsafe(mrsRemoteVideoTrackHandle videoTrackHandle) {
+  auto existing = NativeRenderer::GetUnsafe(videoTrackHandle);
+  if (existing) {
+    existing->Shutdown();
+    g_renderers.erase(videoTrackHandle);
+  }
+}
+
+std::shared_ptr<NativeRenderer> NativeRenderer::Get(mrsRemoteVideoTrackHandle videoTrackHandle) {
+  {
+    // Global lock
+    std::lock_guard guard(g_lock);
+    return GetUnsafe(videoTrackHandle);
+  }
+}
+
+std::shared_ptr<NativeRenderer> NativeRenderer::GetUnsafe(mrsRemoteVideoTrackHandle videoTrackHandle) {
+  auto iter = g_renderers.find(videoTrackHandle);
+  if (iter == g_renderers.end())
+    return nullptr;
+  else
+    return iter->second;
+}
+
+std::vector<std::shared_ptr<NativeRenderer>> NativeRenderer::MultiGetUnsafe(const std::set<mrsRemoteVideoTrackHandle>& videoTrackHandles) {
+  std::vector<std::shared_ptr<NativeRenderer>> renderers;
+  for (auto peerHandle : videoTrackHandles) {
+    auto renderer = GetUnsafe(peerHandle);
+    if (renderer) {
+      renderers.push_back(renderer);
+    }
+  }
+  return std::move(renderers);
+}
+
+NativeRenderer::NativeRenderer(mrsRemoteVideoTrackHandle videoTrackHandle) : m_handle(videoTrackHandle) {
+  Log_Debug("NativeRenderer::NativeRenderer");
+  if (!g_renderApi) {
+    Log_Warning("NativeRenderer: Unity plugin not initialized.");
+  }
+}
+
+NativeRenderer::~NativeRenderer() {
+  /// This is unsafe to log in Unity Editor.
+  // Log_Debug("NativeRenderer::~NativeRenderer");
+}
+
+void NativeRenderer::Shutdown() {
+  Log_Debug("NativeRenderer::Shutdown");
+  DisableRemoteVideo();
+}
+
+void NativeRenderer::EnableRemoteVideo(mrsRemoteVideoTrackHandle videoTrackHandle,
+                                        VideoKind format,
+                                        TextureDesc textureDescs[],
+                                        int textureDescCount) {
+  if (!g_renderApi) {
+    Log_Warning("NativeRenderer: Unity plugin not initialized.");
+  }
+  {
+    // Instance lock
+    std::lock_guard guard(m_lock);
+    m_remoteVideoFormat = format;
+    switch (format) {
+      case VideoKind::kI420:
+        if (textureDescCount == 3) {
+          m_remoteTextures.resize(3);
+          m_remoteTextures[0] = textureDescs[0];
+          m_remoteTextures[1] = textureDescs[1];
+          m_remoteTextures[2] = textureDescs[2];
+          mrsRemoteVideoTrackRegisterI420AFrameCallback(
+              videoTrackHandle, NativeRenderer::I420ARemoteVideoFrameCallback,
+              m_handle);
+        }
+        break;
+
+      case VideoKind::kARGB:
+        // TODO
+        break;
+
+      case VideoKind::kNone:
+        // TODO
+        break;
```

Comment:
These should return a failure through the API, and/or there should be a comment on the API function saying that the only supported format is I420.

Reply:
Added warning logs and note to api summary.

Reply:
Can you add a comment to `NativeVideoRenderer` and `mrsNativeRenderer_EnableRemoteVideo` saying that only I420A is supported at the moment?

Reply:
Logged support for ARGB as #632